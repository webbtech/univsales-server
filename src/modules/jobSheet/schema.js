import { gql } from 'apollo-server-lambda'

export default gql`
extend type Mutation {
  jobSheetDuplicate(id: ID!): JobSheet
  jobSheetPersist(addressInput: AddressInput, jobSheetInput: JobSheetInput!): JobSheet
  jobSheetPersistFeatures(features: String!, id: ID!): JobSheet
  jobSheetPersistGroup(input: GroupInput!): JobSheetGroup
  jobSheetPersistOther(input: OtherInput!): JobSheetOther
  jobSheetPersistWindow(input: WindowInput!): JobSheetWindow
  jobSheetRemove(id: ID!): DBResult
  jobSheetRemoveGroup(id: ID!): DBResult
  jobSheetRemoveOther(id: ID!): DBResult
  jobSheetRemoveWindow(id: ID!): DBResult
}
extend type Query {
  jobSheetData(jobSheetID: ID!): JobSheetData
  jobSheetGroup(groupID: ID!): JobSheetGroup
  jobSheetOther(otherID: ID!): JobSheetOther
  jobSheetWindow(windowID: ID!): JobSheetWindow
  searchJobSheetsByCustomer(customerID: ID!): [JobSheet]
}

#
## ================================== Types ==================================
#

type GroupItemCosts {
  extendTotal: Float
  extendUnit: Float
}

type GroupItemDims {
  height: WindowDims
  width: WindowDims
}

type GroupItemSpecs {
  extendSqft: Int
  options: String
  overSize: Int
  sqft: Int
}

type GroupWindowDims {
  decimal: Float
  diff: Float
  fraction: String
  inch: Int
}

type JobSheet {
  _id: ID!
  addressID: Address
  duplicateNumber: Int
  createdAt: String
  customerID: Customer
  features: String
  number: Int!
  updatedAt: String
}

type JobSheetData {
  groups: [JobSheetGroup]
  jobsheet: JobSheet
  other: [JobSheetOther]
  windows: [JobSheetWindow]
}

type JobSheetGroup {
  _id: ID!
  costs: JobSheetGroupCosts
  createdAt: String
  dims: JobSheetGroupDims
  items: [JobSheetGroupItem]
  jobsheetID: ID!
  qty: Int
  rooms: [String]
  specs: JobSheetGroupSpecs
  updatedAt: String
}

type JobSheetGroupCosts {
  discounted: Float
  discountedAmount: Float
  extendTotal: Float
  extendUnit: Float
  install: Float
  installType: Float
  netUnit: Float
  options: Float
  trim: Float
  windows: Float
}

type JobSheetGroupDims {
  height: GroupWindowDims
  width: GroupWindowDims
}

type JobSheetGroupItem {
  _id: ID
  costs: GroupItemCosts
  dims: GroupItemDims
  product: Product
  productID: ID
  qty: Int
  specs: GroupItemSpecs
}

type JobSheetGroupSpecs {
  groupTypeDescription: String
  installType: String
  options: String
  optionsDoc: OptionsDoc
  sqft: Int
  style: String
  trim: String
}

type JobSheetItemCosts {
  discounted: Float
  discountedAmount: Float
  extendTotal: Float
  extendUnit: Float
  install: Float
  installType: Float
  netUnit: Float
  options: Float
  trim: Float
  window: Float
}

type JobSheetOther {
  _id: ID!
  costs: OtherCosts
  createdAt: String
  description: String
  jobsheetID: ID
  product: String
  qty: Int
  rooms: [String]
  specs: OtherSpecs
  updatedAt: String
}

type JobSheetWindow {
  _id: ID!
  brickmould: BrickMould
  costs: JobSheetItemCosts
  createdAt: String
  dims: JobSheetWindowDims
  jobsheetID: ID
  productID: Product
  qty: Int
  rooms: [String]
  specs: JobSheetWindowSpecs
  updatedAt: String
}

type JobSheetWindowDims {
  height: WindowDims
  width: WindowDims
}

type JobSheetWindowSpecs {
  installType: String
  options: String
  optionsDoc: OptionsDoc
  overSize: Int
  sqft: Int
  trim: String
}

type OtherCosts {
  extendTotal: Float
  extendUnit: Float
}

type OtherSpecs {
  location: String
  options: String
}

type WindowDims {
  decimal: Float
  fraction: String
  inch: Int
  overSize: Int
  round: Int
  underSize: Int
}

type OptionsDoc {
  color: String
  customColor: String
  darker: Boolean
  extraDetails: String
  window: [String]
}

type BrickMould {
  dims: BrickMouldDims
  size: Int
  specs: String
}

type BrickMouldDim {
  decimal: Float
  inch: Int
  fraction: String
}

type BrickMouldDims {
  height: BrickMouldDim
  width: BrickMouldDim
}

#
## ================================== Inputs ==================================
#

input GroupCostsInput {
  discountAmount: Float
  discounted: Float
  extendTotal: Float
  extendUnit: Float
  install: Float
  installType: Float
  netUnit: Float
  options: Float
  trim: Float
  windows: Float
}

input GroupDimsInput {
  decimal: Float
  diff: Float
  fraction: String
  inch: Int
}

input GroupHWDimsInput {
  height: GroupDimsInput
  width: GroupDimsInput
}

input GroupInput {
  _id: ID
  costs: GroupCostsInput
  dims: GroupHWDimsInput
  items: [GroupItemInput]
  jobsheetID: ID!
  qty: Int
  rooms: [String]
  specs: GroupSpecsInput
}

input GroupItemCostsInput {
  extendTotal: Float
  extendUnit: Float
}

input GroupItemInput {
  _id: ID!
  costs: GroupItemCostsInput
  dims: WindowHWDimsInput
  product: GroupItemProductInput
  productID: ID!
  qty: Int
  specs: GroupItemSpecsInput
}

input GroupItemProductInput {
  name: String
}

input GroupItemSpecsInput {
  extendSqft: Int
  options: String
  overSize: Int
  sqft: Int
}

input GroupSpecsInput {
  groupTypeDescription: String
  installType: String
  options: String
  optionsDoc: OptionsDocInput
  sqft: Int
  style: String
  trim: String
}

input JobSheetInput {
  _id: ID
  customerID: ID!
  features: String
}

input OtherCostsInput {
  extendTotal: Float
  extendUnit: Float
}

input OtherInput {
  _id: ID
  costs: OtherCostsInput
  description: String
  jobsheetID: ID
  product: String
  qty: Int
  rooms: [String]
  specs: OtherSpecsInput
}

input OtherSpecsInput {
  location: String
  options: String
}

input WindowCostsInput {
  discounted: Float
  discountedAmount: Float
  extendTotal: Float
  extendUnit: Float
  install: Float
  installType: Float
  netUnit: Float
  options: Float
  trim: Float
  window: Float
}

input WindowDimsInput {
  decimal: Float
  fraction: String
  inch: Int
  overSize: Int
  round: Int
  underSize: Int
}

input WindowHWDimsInput {
  height: WindowDimsInput
  width: WindowDimsInput
}

input WindowInput {
  _id: ID
  costs: WindowCostsInput
  dims: WindowHWDimsInput
  jobsheetID: ID!
  productID: ID!
  qty: Int
  rooms: [String]
  specs: WindowSpecsInput
}

input WindowSpecsInput {
  extendSqft: Int
  installType: String
  options: String
  optionsDoc: OptionsDocInput
  overSize: Int
  sqft: Int
  trim: String
}

input OptionsDocInput {
  color: String
  customColor: String
  darker: Boolean
  extraDetails: String
  window: [String]
}

input BrickMouldInput {
  dims: BrickMouldDimsInput
  size: Int
  specs: String
}
input BrickMouldDimsInput {
  height: BrickMouldDimInput
  width: BrickMouldDimInput
}

input BrickMouldDimInput {
  decimal: Float
  fraction: String
  inch: Int
}
`
