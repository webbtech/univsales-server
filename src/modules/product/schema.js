import { gql } from 'apollo-server'

export default gql`
extend type Mutation {
  productPersist(input: ProductInput!): Product
}

extend type Query {
  product(productID: ID!): Product
  products: [Product]
}

type Product {
  _id: ID!
  maxHeight: Int
  maxWidth: Int
  minHeight: Int
  minWidth: Int
  name: String!
  premium: ProductPremium
  sizeCost: JSON
}

type ProductPremium {
  cost: Int
  oversizeLimit: Int
}

input ProductInput {
_id: ID!
  maxHeight: Int!
  maxWidth: Int!
  minHeight: Int!
  minWidth: Int!
  name: String!
  premium: ProductPremiumInput!
  sizeCost: JSON!
}

input ProductPremiumInput {
  cost: Int!
  oversizeLimit: Int!
}
`
