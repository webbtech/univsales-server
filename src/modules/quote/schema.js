import { gql } from 'apollo-server'

export default gql`
extend type Mutation {
  createInvoice(id: ID!): Quote
  createPDF(input: PDFInput!): DBResult
  quotePersist(input: QuoteInput!): Quote
  quotePersistDiscount(input: QuoteDiscountInput!): Quote
  quoteRemove(id: ID!): DBResult
}

extend type Query {
  quote(quoteID: ID!): Quote
  quoteNearbyJobs(input: AddressGeoInput): [AddressGeo]
  pdfSignedURL(input: SignedURLInput!): PDFSignedURL
  searchCustomerRecent: [QuoteRecent]
  searchQuotes(closed: Boolean, invoiced: Boolean, year: String): Quotes
  searchQuotesByCustomer(customerID: ID!): Quotes
}

type PDFSignedURL {
  code: Int
  data: SignedURLData
  message: String
  status: String
  timestamp: Int
}

type Quote {
  _id: ID!
  closed: Boolean!
  createdAt: String
  customerID: Customer
  # deposit: Boolean!
  discount: QuoteDiscount
  invoiced: Boolean!
  itemCosts: QuoteItemCosts
  itemSummary: QuoteItemSummary
  items: QuoteItems
  jobsheetID: JobSheet
  number: Int!
  quotePrice: QuotePrice
  updatedAt: String
  version: Int!
}

type QuoteDiscount {
  description: String
  discount: Float
  subtotal: Float
  tax: Float
  total: Float
}

type QuoteItemCosts {
  group: Float
  other: Float
  subtotal: Float
  window: Float
}

type QuoteItemSummary {
  group: QuoteItemSummaryDetails
  other: QuoteItemSummaryDetails
  window: QuoteItemSummaryDetails
}

type QuoteItemSummaryCosts {
  extendTotal: Float
  extendUnit: Float
  netUnit: Float
}

type QuoteItemSummaryDetails {
  items: [QuoteItemSummaryItems]
  totals: QuoteItemSummaryExtendTotal
}

type QuoteItemSummaryExtendTotal {
  extendTotal: Float
}

type QuoteItemSummaryItems {
  costs: QuoteItemSummaryCosts
  description: String
  qty: Int
  rooms: String
  specs: JSON
}

type QuoteItems {
  group: [String]
  other: [String]
  window: [String]
}

type QuotePrice {
  outstanding: Float
  payments: Float
  subtotal: Float
  tax: Float
  total: Float
}

type QuoteRecent {
  _id: ID!
  address: QuoteRecentAddress
  customer: Customer
  customerID: ID
  invoiced: Boolean
  number: Int
  quotePrice: QuotePrice
  updatedAt: String
  version: Int
}

type QuoteRecentAddress {
  _id: ID
  city: String
  street1: String
}

type Quotes {
  quotes: [Quote]!
  totalInvoiced: Float
  totalOutstanding: Float
}

type SignedURLData {
  url: String
}

#
## ================================== Inputs ==================================
#

input DiscountInput {
  description: String
  discount: Float
  subtotal: Float
  tax: Float
  total: Float
}

input ItemCostsInput {
  group: Float
  other: Float
  subtotal: Float
  window: Float
}

input ItemSummaryCostsInput {
  extendTotal: Float
  extendUnit: Float
  netUnit: Float
}

input ItemSummaryDetailsInput {
  items: [ItemSummaryItemsInput]
  totals: ItemSummaryExtendTotalInput
}

input ItemSummaryExtendTotalInput {
  extendTotal: Float
}

input ItemSummaryInput {
  group: ItemSummaryDetailsInput
  other: ItemSummaryDetailsInput
  window: ItemSummaryDetailsInput
}

input ItemSummaryItemsInput {
  costs: ItemSummaryCostsInput
  description: String
  qty: Int
  rooms: String
  specs: JSON
}

input ItemsInput {
  group: [String]
  other: [String]
  window: [String]
}

input PDFInput {
  quoteID: ID!
  docType: String
}

input QuoteDiscountInput {
  _id: ID!
  discount: DiscountInput!
  quotePrice: QuotePriceInput!
}

input QuoteInput {
  _id: ID
  customerID: ID!
  discount: DiscountInput
  features: String
  itemCosts: ItemCostsInput
  itemSummary: ItemSummaryInput
  items: ItemsInput
  jobsheetID: ID!
  quotePrice: QuotePriceInput
  version: Int
}

input QuotePriceInput {
  outstanding: Float
  payments: Float
  subtotal: Float
  tax: Float
  total: Float
}

input SignedURLInput {
  number: Int!
  type: String!
  version: Int
}
`
