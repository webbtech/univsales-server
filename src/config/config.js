import fs from 'fs'
import path from 'path'
import yaml from 'js-yaml'
import AWS from 'aws-sdk'

// see https://hackernoon.com/you-should-use-ssm-parameter-store-over-lambda-env-variables-5197fc6ea45b
// for example using expiry with events

const config = {
  defaults: {},
  defaultsFN: 'defaults.yml',
  defaultsPath: '',
  ssmPath: '',
  config: {},
}

config.load = async function load() {
  try {
    this.setDefaults()
    await this.loadSSM()
    return this.config
  } catch (e) {
    throw new Error(e)
  }
}

config.setDefaultsPath = function setDefaultsPath(filename = this.defaultsFN) {
  this.defaultsPath = path.join(__dirname, filename)
  try {
    fs.accessSync(this.defaultsPath, fs.F_OK)
  } catch (e) {
    // console.error(e) // eslint-disable-line
    throw new Error(e)
  }
}

config.setDefaults = function setDefaults() {
  if (!this.defaultsPath) {
    this.setDefaultsPath()
  }
  try {
    this.defaults = yaml.safeLoad(fs.readFileSync(this.defaultsPath, 'utf8'))
  } catch (e) {
    // console.error(e) // eslint-disable-line
    throw new Error(e)
  }

  // Ensure we have ssm path
  if (this.defaults.usessm && !this.defaults.ssmPath) {
    throw new Error('Missing ssmPath in defaults')
  }

  // set stage
  this.defaults.nodeEnv = process.env.Stage ? process.env.Stage : process.env.NODE_ENV
}

config.getDefaults = function getDefaults() {
  return this.defaults
}

config.setSSMPath = function setSSMPath() {
  if (!this.defaults.ssmPath || !this.defaults.awsRegion) {
    throw new Error('Missing awsRegion, stage or ssmPath in defaults')
  }

  switch (this.defaults.nodeEnv) {
    case 'test':
    case 'dev':
    case 'development':
      this.ssmPath = `/test/${this.defaults.ssmPath}`
      break
    default:
      this.ssmPath = `/${this.defaults.nodeEnv}/${this.defaults.ssmPath}`
  }
}

config.loadSSM = async function loadSSM() {
  if (!this.defaults.usessm || !this.defaults.ssmPath) {
    console.log('loadSSM called without usessm or ssmPath variables set') // eslint-disable-line
    return false
  }
  this.setSSMPath()

  const ssm = new AWS.SSM({
    apiVersion: '2014-11-06',
    region: this.defaults.awsRegion,
  })

  const ssmParams = {
    Path: this.ssmPath,
    WithDecryption: true,
  }

  let params
  try {
    params = await ssm.getParametersByPath(ssmParams).promise()
  } catch (e) {
    throw new Error(e)
  }

  const ssmVals = {}
  params.Parameters.forEach((p) => {
    ssmVals[p.Name.split('/')[3]] = p.Value
  })

  this.config = Object.assign({}, this.defaults, ssmVals) // eslint-disable-line
  // this.config = { ...this.defaults, ...ssmVals } // this, oddly, doesn't compile

  return true
}

module.exports = config
