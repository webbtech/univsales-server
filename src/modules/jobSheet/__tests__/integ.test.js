import { createTestClient } from 'apollo-server-testing'
import { ApolloServer, gql } from 'apollo-server'

import schema from '../../../schema'
import config from '../../../config/config'
import db from '../../../mongo/connect'

import {
  address,
  jobsheet,
  features,
  groupInput,
  otherInput,
  windowInput,
} from '../mockData'

// test with: yarn test:w src/modules/jobSheet

const { typeDefs, resolvers } = schema
let query
let mutate
let newJobSheetID
let newGroupID
let newOtherID
let newWindowID

const customerID = '5b2122022aac043b0c7ec06d'
const jobSheetID = '5b1846d52aac0450227ebfe9'

const JOBSHEET_DATA = gql`
query GetJobSheetData($jobSheetID: ID!) {
  jobSheetData(jobSheetID: $jobSheetID) {
    jobsheet {
      __typename
      _id
      createdAt
      updatedAt
      features
      addressID {
        street2
        _id
        street1
        city
      }
      customerID {
        __typename
        _id
        name {
          first
          last
          spouse
        }
      }
    }
    windows {
      __typename
      _id
      costs {
        __typename
        extendTotal
        extendUnit
        install
        installType
        netUnit
        options
        trim
        window
      }
      qty
      rooms
      createdAt
      updatedAt
    }
    groups {
      __typename
      _id
      costs {
        __typename
        extendTotal
        extendUnit
        install
        installType
        netUnit
        options
        trim
        windows
      }
      qty
      rooms
      specs {
        __typename
        groupTypeDescription
      }
    }
    other {
      __typename
      _id
      costs {
        __typename
        extendTotal
        extendUnit
      }
      description
      qty
      rooms
      specs {
        __typename
        options
        location
      }
      updatedAt
      createdAt
    }
  }
}`
const CUSTOMER_JOBSHEETS = gql`
query JobSheetsByCustomer($customerID: ID!) {
  searchJobSheetsByCustomer(customerID: $customerID) {
    _id
    addressID {
      _id
      street1
      city
    }
    number
    updatedAt
  }
}`

const PERSIST_JOBSHEET = gql`
mutation jobSheetPersist(
  $jobSheetInput: JobSheetInput!,
  $addressInput: AddressInput,
) {
  jobSheetPersist(
    jobSheetInput: $jobSheetInput,
    addressInput: $addressInput,
  ) {
    _id
  }
}`

const REMOVE_JOBSHEET = gql`
mutation jobSheetRemove($id: ID!) {
  jobSheetRemove(id: $id) {
    n
    ok
  }
}`

const PERSIST_FEATURES = gql`
mutation jobSheetPersistFeatures($id: ID!, $features: String!) {
  jobSheetPersistFeatures(id: $id, features: $features) {
    _id
    features
  }
}`

const PERSIST_GROUP = gql`
mutation jobSheetPersistGroup($input: GroupInput!) {
  jobSheetPersistGroup(input: $input) {
    _id
  }
}`

const REMOVE_GROUP = gql`
mutation jobSheetRemoveGroup($id: ID!) {
  jobSheetRemoveGroup(id: $id) {
    n
    ok
  }
}`

const PERSIST_OTHER = gql`
mutation jobSheetPersistOther($input: OtherInput!) {
  jobSheetPersistOther(input: $input) {
    _id
  }
}`

const REMOVE_OTHER = gql`
mutation jobSheetRemoveOther($id: ID!) {
  jobSheetRemoveOther(id: $id) {
    n
    ok
  }
}`

const PERSIST_WINDOW = gql`
mutation jobSheetPersistWindow($input: WindowInput!) {
  jobSheetPersistWindow(input: $input) {
    _id
  }
}`

const REMOVE_WINDOW = gql`
mutation jobSheetRemoveWindow($id: ID!) {
  jobSheetRemoveWindow(id: $id) {
    n
    ok
  }
}`

let dbConn
beforeAll(async () => {
  const cfg = await config.load()
  dbConn = await db.connect(cfg)

  const server = new ApolloServer({
    typeDefs,
    resolvers,
  })
  const testClient = createTestClient(server)
  query = testClient.query // eslint-disable-line prefer-destructuring
  mutate = testClient.mutate // eslint-disable-line prefer-destructuring
})

afterAll(async () => {
  dbConn.close()
})

test('jobSheetData', async () => {
  const res = await query({ query: JOBSHEET_DATA, variables: { jobSheetID } })
  const { jobSheetData } = res.data
  expect(jobSheetData).toBeTruthy()
  // expect(res).toMatchSnapshot()
})

test('searchJobSheetsByCustomer', async () => {
  const res = await query({ query: CUSTOMER_JOBSHEETS, variables: { customerID } })
  const { searchJobSheetsByCustomer } = res.data
  expect(searchJobSheetsByCustomer).toBeTruthy()
  // expect(res).toMatchSnapshot()
})

describe('persist mutations', () => {
  afterAll(async () => {
    if (newJobSheetID) {
      const res = await mutate({
        mutation: REMOVE_JOBSHEET,
        variables: { id: newJobSheetID },
      })
      const { jobSheetRemove } = res.data
      expect(jobSheetRemove.n).toEqual(1)
      expect(jobSheetRemove.ok).toEqual(1)
    }

    if (newGroupID) {
      const res = await mutate({
        mutation: REMOVE_GROUP,
        variables: { id: newGroupID },
      })
      const { jobSheetRemoveGroup } = res.data
      expect(jobSheetRemoveGroup.n).toEqual(1)
      expect(jobSheetRemoveGroup.ok).toEqual(1)
    }

    if (newOtherID) {
      const res = await mutate({
        mutation: REMOVE_OTHER,
        variables: { id: newOtherID },
      })
      const { jobSheetRemoveOther } = res.data
      expect(jobSheetRemoveOther.n).toEqual(1)
      expect(jobSheetRemoveOther.ok).toEqual(1)
    }

    if (newWindowID) {
      const res = await mutate({
        mutation: REMOVE_WINDOW,
        variables: { id: newWindowID },
      })
      const { jobSheetRemoveWindow } = res.data
      expect(jobSheetRemoveWindow.n).toEqual(1)
      expect(jobSheetRemoveWindow.ok).toEqual(1)
    }
  })

  test('persistJobsheet', async () => {
    const res = await mutate({
      mutation: PERSIST_JOBSHEET,
      variables: { addressInput: address, jobSheetInput: jobsheet },
    })
    const { jobSheetPersist } = res.data
    newJobSheetID = jobSheetPersist._id
    expect(jobSheetPersist).toBeTruthy()
  })

  test('jobSheetPersistFeatures', async () => {
    const res = await mutate({
      mutation: PERSIST_FEATURES,
      variables: { id: newJobSheetID, features },
    })
    const { jobSheetPersistFeatures: jobSheet } = res.data
    expect(jobSheet.features).toEqual(features)
  })

  test('jobSheetPersistGroup', async () => {
    const res = await mutate({
      mutation: PERSIST_GROUP,
      variables: { input: groupInput },
    })
    const { jobSheetPersistGroup } = res.data
    newGroupID = jobSheetPersistGroup._id
    expect(jobSheetPersistGroup).toBeTruthy()
  })

  test('jobSheetPersistOther', async () => {
    const res = await mutate({
      mutation: PERSIST_OTHER,
      variables: { input: otherInput },
    })
    const { jobSheetPersistOther } = res.data
    newOtherID = jobSheetPersistOther._id
    expect(jobSheetPersistOther).toBeTruthy()
  })

  test('jobSheetPersistWindow', async () => {
    const res = await mutate({
      mutation: PERSIST_WINDOW,
      variables: { input: windowInput },
    })
    const { jobSheetPersistWindow } = res.data
    newWindowID = jobSheetPersistWindow._id
    expect(jobSheetPersistWindow).toBeTruthy()
  })
})
