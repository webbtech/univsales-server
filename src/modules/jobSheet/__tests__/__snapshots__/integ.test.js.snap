// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jobSheetData 1`] = `
Object {
  "data": Object {
    "jobSheetData": Object {
      "groups": Array [
        Object {
          "__typename": "JobSheetGroup",
          "_id": "5b19df4a2aac04041e7ebff5",
          "costs": Object {
            "__typename": "JobSheetGroupCosts",
            "extendTotal": 2010,
            "extendUnit": 2010,
            "install": 600,
            "installType": 300,
            "netUnit": 2010,
            "options": 150,
            "trim": 100,
            "windows": 860,
          },
          "qty": 1,
          "rooms": Array [
            "BR",
          ],
          "specs": Object {
            "__typename": "JobSheetGroupSpecs",
            "groupTypeDescription": "2 Casements",
          },
        },
        Object {
          "__typename": "JobSheetGroup",
          "_id": "5b1a70802aac04956a7ec029",
          "costs": Object {
            "__typename": "JobSheetGroupCosts",
            "extendTotal": 3950,
            "extendUnit": 3950,
            "install": 1000,
            "installType": 500,
            "netUnit": 3950,
            "options": 250,
            "trim": 500,
            "windows": 1700,
          },
          "qty": 1,
          "rooms": Array [
            "DR",
          ],
          "specs": Object {
            "__typename": "JobSheetGroupSpecs",
            "groupTypeDescription": "Bow Window 2 Casements, 3 Fixed",
          },
        },
        Object {
          "__typename": "JobSheetGroup",
          "_id": "5b1a6fe92aac0423f37ec022",
          "costs": Object {
            "__typename": "JobSheetGroupCosts",
            "extendTotal": 1830,
            "extendUnit": 1830,
            "install": 500,
            "installType": 250,
            "netUnit": 1830,
            "options": 120,
            "trim": 100,
            "windows": 860,
          },
          "qty": 1,
          "rooms": Array [
            "BR",
          ],
          "specs": Object {
            "__typename": "JobSheetGroupSpecs",
            "groupTypeDescription": "2 Casements",
          },
        },
        Object {
          "__typename": "JobSheetGroup",
          "_id": "5b195db0c2e75ffe21fdd0ed",
          "costs": Object {
            "__typename": "JobSheetGroupCosts",
            "extendTotal": 3450,
            "extendUnit": 3450,
            "install": 1000,
            "installType": 500,
            "netUnit": 3450,
            "options": 150,
            "trim": 100,
            "windows": 1700,
          },
          "qty": 1,
          "rooms": Array [
            "DR",
          ],
          "specs": Object {
            "__typename": "JobSheetGroupSpecs",
            "groupTypeDescription": "Bow Window 2 Casements, 3 Fixed",
          },
        },
        Object {
          "__typename": "JobSheetGroup",
          "_id": "603d3575869503690158e2b5",
          "costs": Object {
            "__typename": "JobSheetGroupCosts",
            "extendTotal": 1970,
            "extendUnit": 1970,
            "install": 775,
            "installType": 0,
            "netUnit": 1970,
            "options": 0,
            "trim": 0,
            "windows": 1195,
          },
          "qty": 1,
          "rooms": Array [
            "LR",
          ],
          "specs": Object {
            "__typename": "JobSheetGroupSpecs",
            "groupTypeDescription": "2 Double Hungs- 1 Fixed ",
          },
        },
      ],
      "jobsheet": Object {
        "__typename": "JobSheet",
        "_id": "5b1846d52aac0450227ebfe9",
        "addressID": Object {
          "_id": "5b1846d52aac04948a7ebfe8",
          "city": "Welland",
          "street1": "18 Chantler Rd",
          "street2": null,
        },
        "createdAt": "1528317653405",
        "customerID": Object {
          "__typename": "Customer",
          "_id": "5b1846c62aac040faf7ebfe7",
          "name": Object {
            "first": "Joe",
            "last": "DiMarco (tester)",
            "spouse": "",
          },
        },
        "features": "A.C. Vinyl Windows (Darker)
Low E & Argon Gas
Full-frame Change 
2 3/4\\" Fingerjoint Trim
Aluminum Capping
Complete Garbage Removal
Installed Including HST",
        "updatedAt": "1528422516019",
      },
      "other": Array [
        Object {
          "__typename": "JobSheetOther",
          "_id": "5b19e00f2aac04230d7ec012",
          "costs": Object {
            "__typename": "OtherCosts",
            "extendTotal": 255,
            "extendUnit": 255,
          },
          "createdAt": "1528422415342",
          "description": "Cap Garage Frame
Ok",
          "qty": 1,
          "rooms": Array [
            "BR",
          ],
          "specs": Object {
            "__typename": "OtherSpecs",
            "location": "Driveway",
            "options": "Pebble Colour",
          },
          "updatedAt": "1582571391922",
        },
        Object {
          "__typename": "JobSheetOther",
          "_id": "603d3575869503690158e2b8",
          "costs": Object {
            "__typename": "OtherCosts",
            "extendTotal": 825,
            "extendUnit": 825,
          },
          "createdAt": "1614624117165",
          "description": "Installation of Single S/L Door
Includes Aluminum Capping",
          "qty": 1,
          "rooms": Array [
            "BR",
          ],
          "specs": Object {
            "__typename": "OtherSpecs",
            "location": "",
            "options": "",
          },
          "updatedAt": "1614624117165",
        },
      ],
      "windows": Array [
        Object {
          "__typename": "JobSheetWindow",
          "_id": "5b18474a2aac045b127ebfeb",
          "costs": Object {
            "__typename": "JobSheetItemCosts",
            "extendTotal": 1185,
            "extendUnit": 1185,
            "install": 300,
            "installType": 200,
            "netUnit": 1185,
            "options": 60,
            "trim": 75,
            "window": 550,
          },
          "createdAt": "1528317732135",
          "qty": 1,
          "rooms": Array [
            "BR",
          ],
          "updatedAt": "1582571334118",
        },
        Object {
          "__typename": "JobSheetWindow",
          "_id": "5b1847242aac0405487ebfea",
          "costs": Object {
            "__typename": "JobSheetItemCosts",
            "extendTotal": 1085,
            "extendUnit": 1085,
            "install": 300,
            "installType": 200,
            "netUnit": 1085,
            "options": 60,
            "trim": 75,
            "window": 450,
          },
          "createdAt": "1528317732135",
          "qty": 1,
          "rooms": Array [
            "BR",
          ],
          "updatedAt": "1528317755342",
        },
        Object {
          "__typename": "JobSheetWindow",
          "_id": "5e541f55aaa6490007efcede",
          "costs": Object {
            "__typename": "JobSheetItemCosts",
            "extendTotal": 415,
            "extendUnit": 415,
            "install": 0,
            "installType": 0,
            "netUnit": 415,
            "options": 0,
            "trim": 0,
            "window": 415,
          },
          "createdAt": "1582571349294",
          "qty": 1,
          "rooms": Array [],
          "updatedAt": "1582571349294",
        },
      ],
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`searchJobSheetsByCustomer 1`] = `
Object {
  "data": Object {
    "searchJobSheetsByCustomer": Array [
      Object {
        "_id": "5ce69196ec2d8e7c1cb03dec",
        "addressID": Object {
          "_id": "5ce69196ec2d8ee6f5b03deb",
          "city": "Niagara-on-the-Lake",
          "street1": "44 Queen St",
        },
        "number": 1321,
        "updatedAt": "1558614422846",
      },
      Object {
        "_id": "5b2122152aac04d9f57ec06f",
        "addressID": Object {
          "_id": "5b2122152aac0418527ec06e",
          "city": "Hamilton",
          "street1": "Hamilton St",
        },
        "number": 1231,
        "updatedAt": "1554995207743",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
